diff --git a/rtl/if_c_obi.sv b/rtl/if_c_obi.sv
index 152715e..29a0a05 100644
--- a/rtl/if_c_obi.sv
+++ b/rtl/if_c_obi.sv
@@ -24,11 +24,14 @@
 // 
 // The 'c' in the interface names means 'Compressed'
 // since this interface is a subset of the full OBI spec.
-interface if_c_obi import cv32e40x_pkg::*;
+interface if_c_obi
 #(
-    parameter type REQ_TYPE  = obi_inst_req_t,
-    parameter type RESP_TYPE = obi_inst_resp_t
+    parameter type REQ_TYPE  = cv32e40x_pkg::obi_inst_req_t,
+    parameter type RESP_TYPE = cv32e40x_pkg::obi_inst_resp_t
 );
+
+    import cv32e40x_pkg::*;
+
     // A channel signals
     obi_req_t                  s_req;
     obi_gnt_t                  s_gnt;
diff --git a/rtl/if_core_v_xif.sv b/rtl/if_core_v_xif.sv
index a701932..6fd1b45 100644
--- a/rtl/if_core_v_xif.sv
+++ b/rtl/if_core_v_xif.sv
@@ -28,7 +28,7 @@
 //                                                                            //
 ////////////////////////////////////////////////////////////////////////////////
 
-interface if_xif import cv32e40x_pkg::*;
+interface if_xif
 #(
   parameter int          X_NUM_RS        =  2,  // Number of register file read ports that can be used by the eXtension interface
   parameter int          X_MEM_WIDTH     =  32, // Memory access width for loads/stores via the eXtension interface
@@ -37,6 +37,8 @@ interface if_xif import cv32e40x_pkg::*;
   parameter logic [31:0] X_MISA          =  '0  // MISA extensions implemented on the eXtension interface
 );
 
+  import cv32e40x_pkg::*;
+
   localparam int XLEN = 32;
   localparam int FLEN = 32;
 
